Juhitha - Cleanup the inventory types page

Juhitha-Reddy opened the pull request:
# Description
Cleanup and refinement of the All Inventory Types page functionality, building upon the work done in PR #[3540](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3540). This PR addresses remaining issues and improves the implementation of inventory type management features originally developed by Shuhua and later updated by Rahul.
Fixes critical import path issues, API endpoint configuration, and ensures robust communication with the backend inventory management system.

## Related PRS (if any):
This frontend PR is related to the #[765](https://github.com/OneCommunityGlobal/HGNRest/pull/765) backend PR.
To test this frontend PR you need to checkout the #[765](https://github.com/OneCommunityGlobal/HGNRest/pull/765) backend PR.

## Main changes explained:
- Updated all the inventory tables to implement APIs to each
- Implemented edit functionality for all the tables
- Fixed issues with existing issues with add/edit/delete functionalities for all inventory types
- Fixed import path issues in TypesTable.jsx and TypeRow.jsx - changed from absolute to relative paths
- Integrated inventory APIs with proper error handling and toast notifications

## How to test:

1. Check into current branch Cleanup/shuhua-inventory-types-page
2. Do npm install && npm run start:local to run this PR locally
3. Ensure backend is running.
4. Clear site data/cache
5. Go to dashboard → Building Management → Inventory Types(/bmdashboard/inventorytypes)
6. Verify all inventory type operations work:
7. View different inventory categories (Materials, Tools, Equipment, etc.)
8. Add new inventory types
9. Edit existing inventory types
10. Delete inventory types
11. Verify this feature works in dark mode

NOTE: Please make sure you pointing to local api endpoints (http://localhost:4500/api) by updating in the .env file this application.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/bc4d85f1-78b9-4aa2-81e2-91da8fe1f3ea

Commit:
Integrated Inventory APIs and Added Toasts

Juhitha-Reddy committed:
Fixed routing issues

Juhitha-Reddy committed:
Merge branch 'development' into Cleanup/shuhua-inventory-types-page

Namitha7070 reviewed (changes requested):
Tested this PR. The add feature for all fields(materials, consumables, equipments, etc) are working fine but the edit and delete features are not getting updated.

https://github.com/user-attachments/assets/074260ed-9639-4905-9e09-4cbaec39f482

shashankm9 reviewed (changes requested):
I have tested this PR. Add feature is working in all the Inventory Types But Edit and delete functions are not works in any of those and dark mode is not working at all. Inventory Types are not legible when curious is hovering on then, change the text color to white maybe.

https://github.com/user-attachments/assets/efaba982-8292-4cda-b4c2-ef37a102510b

Juhitha-Reddy commented:
@Rishitha1810 , @shashankm9, @Namitha7070 , I verified the PR in my local and the add, edit and delete functionalities are working as expected. Could you please check if you have updated the api endpoint to local while testing and please let me know if the issue still persists.

Anusha-Gali reviewed (changes requested):
Hi Juhitha,

I have reviewed your PR locally and all the functionality works as expected but as mentioned by you i do not see dark mode implemented for the page.

https://github.com/user-attachments/assets/a479331b-2455-4c7c-afed-f668b8a533d2

Juhitha-Reddy committed:
Merge branch 'development' into Cleanup/shuhua-inventory-types-page

Juhitha-Reddy committed:
updated typelist css

Anusha-Gali reviewed (dismissed):
Hi Juhitha,

Thanks for updating and working on the issue. I have reviewed your PR locally and the dark mode works as per requirement.
<img width="1243" height="793" alt="Screenshot 2026-01-12 at 8 43 26 PM" src="https://github.com/user-attachments/assets/6b0c6f65-0cab-42d0-9f7b-0fda36ce8af8" />
<img width="1243" height="793" alt="Screenshot 2026-01-12 at 9 01 51 PM" src="https://github.com/user-attachments/assets/bb5a9733-09b6-4ff5-b56d-a23d3107631b" />
<img width="1243" height="793" alt="Screenshot 2026-01-12 at 9 02 40 PM" src="https://github.com/user-attachments/assets/3a94a70c-f868-4b31-bc1e-3194675dae07" />
<img width="1243" height="793" alt="Screenshot 2026-01-12 at 9 03 06 PM" src="https://github.com/user-attachments/assets/ab7ab82f-8685-43de-a94d-ec13cb4b5d0e" />

naznin07 reviewed (dismissed):
Dispatch in a connected statement.

<img width="1710" height="320" alt="PR # 3833" src="https://github.com/user-attachments/assets/2c8e5ac0-508b-4d42-99a6-1f9a27acfcb9" />

Juhitha-Reddy committed:
Fix inventory types CRUD operations and resolve merge conflicts

- Resolve merge conflicts in 6 frontend files (TypeRow, TypesTable,
  invUnitsTable, InventoryTypesList, inventoryTypeConstants, TypesList.module.css)
  keeping inline edit/add approach with relative imports
- Add backend route /invtypes/:type/:invtypeId for PUT and DELETE to match
  frontend URL pattern (fixes "Cannot DELETE/PUT" errors)
- Fix Mongoose discriminator issue in updateInventoryType and updateNameAndUnit:
  resolve correct subtype model before updating so discriminator-specific
  fields like unit and fuelType are no longer silently dropped
- Add requestor.requestorId from Redux auth state in addInvType action to
  prevent backend destructuring crash on add operations
- Add Equipment field mapping (desc for description) in addInvType action
- Add addReusableType backend controller and POST /invtypes/reusables route
- Add deleteInventoryType export to invTypeActions for DeleteInvTypeModal
- Merge dark mode styles from both branches in TypesList.module.css

Made-with: Cursor

one-community commented:
Thank you all, merging!
